src/vaadin-progress-mixin.ts

```diff
@@ -0,0 +1,92 @@
+import { Constructor } from '@vaadin/mixin-utils';
+import { LitElement, property, PropertyValues } from 'lit-element';
+
+export interface ProgressMixinInterface {
+  value: number | null | undefined;
+  min: number;
+  max: number;
+  indeterminate: boolean;
+}
+
+export const ProgressMixin = <T extends Constructor<LitElement>>(base: T): T & Constructor<ProgressMixinInterface> => {
+  class Progress extends base {
+    @property({ type: Number })
+    value: number | null | undefined;
+
+    @property({ type: Number })
+    min = 0;
+
+    @property({ type: Number })
+    max = 1;
+
+    @property({ type: Boolean, reflect: true })
+    indeterminate = false;
+
+    protected firstUpdated() {
```

Comment:
Please align this with how `updated` works, to use `super`:

```suggestion
    protected firstUpdated(props: PropertyValues) {
      super.firstUpdated(props);
```